### PR TITLE
[FrameworkBundle][Security] Make CSRF tokens expirable

### PR DESCRIPTION
**Pull request by jderusse**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | TODO

This PR adds an opt-in mechanism to make CSRF tokens expirable (with a different value than session) mitigating replay attacks.

### CODE REVIEW

File: src/Symfony/Component/Security/Csrf/CsrfTokenManager.php [hunk 1]
```diff
@@ -71,8 +73,13 @@ public function getToken(string $tokenId)
         $namespacedId = $this->getNamespace().$tokenId;
         if ($this->storage->hasToken($namespacedId)) {
             $value = $this->storage->getToken($namespacedId);
+            if ($this->isExpired($value)) {
+                $value = (null !== $this->ttl ? time().'.' : '').$this->generator->generateToken();
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Security/Csrf/CsrfTokenManager.php:77]: Won't this create a bad UX? 
Eg the user navigates on the site, gets a token, does something and opens a form 59s after, then submits the form after filling it, which will take more than 1s: its submission is going to be rejected.
Or did I miss something?

Comment (jderusse) [reply src/Symfony/Component/Security/Csrf/CsrfTokenManager.php:77]: oh yeah, you are right.. By design the same token is reused. Meaning an old token with 1 second life could be provided to the user.

**Comment by nicolas-grekas** [comment]
When HTTP is using, we don't care about replay attacks.
When HTTPS is used, replay attacks mean HTTPS has been broken.

I'm not sure this is worth the added complexity.

**Comment by jderusse** [comment]
Replay attacks require to leak a token first. Such leak requires a higher vulnerability (ie. non-HTTPs as mentioned by @nicolas-grekas, or XSS). In such scenario, the protection is worthless.

**Pull request closed without merging**